components/collection/activity/events/Events.vue

```diff
@@ -118,26 +115,18 @@ const getEventAddresses = (event): string[] => {
 
 const eventsAddresses = computed(() => {
   const addresses = props.events.map(getEventAddresses).flat()
-
-  return [...new Set([...addresses])].map(getIdentityId)
+  return [...new Set([...addresses])]
 })
 
-const { data: identities, refetch: getIdentities } = useGraphql({
-  clientName,
-  queryName: 'identities',
-  disabled: computed(() => true),
-  variables: { where: {} },
-})
+const { data: profiles, refresh: refreshProfiles } = useAsyncData(
+  () => fetchProfilesByIds(eventsAddresses.value),
+  { immediate: true },
+)
 
 watch(
   eventsAddresses,
   () => {
-    getIdentities({
-      where: {
-        id_in: eventsAddresses.value,
-        display_isNull: false,
-      },
-    })
+    refreshProfiles()
```

Comment:
https://nuxt.com/docs/api/composables/use-async-data#watch-params


this would be better i think